src/python/WMCore/MicroService/MSTransferor/Workflow.py

```diff
@@ -357,7 +357,7 @@ def getChunkBlocks(self, numChunks=1):
             return [thisChunk], [thisChunkSize]
 
         # create a descendant list of blocks according to their sizes
-        sortedPrimary = sorted(viewitems(self.getPrimaryBlocks()), key=operator.itemgetter(1), reverse=True)
+        sortedPrimary = sorted(viewitems(self.getPrimaryBlocks()), key=operator.itemgetter(0), reverse=True)
```

Comment:
This operator.itemgetter always twists my mind. I had to read the documentation again, and I noticed this change would actually change the expected behavior. Sorting the dictionary on the key, instead of the value.

I have done some checks and this implementation would keep the same behaviour, which is, sorting the dictionary based on the `blockSize` value, reverse order:
```
sorted(viewitems(primaryBlocks), key=lambda item: item[1]['blockSize'], reverse=True)
```

the lambda function might even be more clear. If you agree with this suggestion, you might want to remove the import as well (if it's not used elsewhere).

Just in case, here is the snippet of code I used to test it in py2/py3:
```
from future.utils import viewitems
import operator

primaryBlocks = {'blockC': {'blockSize': 13, 'locations':['siteA', 'siteB']},
                 'blockA': {'blockSize': 11, 'locations':['siteA', 'siteB']},
                 'blockB': {'blockSize': 12, 'locations':['siteA', 'siteB']}}

sorted(viewitems(primaryBlocks), key=operator.itemgetter(0), reverse=True)

sorted(viewitems(primaryBlocks), key=operator.itemgetter(1), reverse=True)

primaryBlocks = {'blockC': {'blockSize': 11, 'locations':['siteA', 'siteB']},
                 'blockA': {'blockSize': 13, 'locations':['siteA', 'siteB']},
                 'blockB': {'blockSize': 12, 'locations':['siteA', 'siteB']}}

sorted(viewitems(primaryBlocks), key=lambda item: item[1]['blockSize'], reverse=True)
```

Reply:
I am sorry, i kept the description and rationale about that change in the PR description, i will add a comment in the "files changed" tab next time.

Thanks soo much for your suggestion, i think I missed entirely the comment `create a descendant list of blocks according to their sizes` and thought that we simply needed a sorted list of the items to speed up later code. 

I think your suggestion greatly helps with both clarity (i had to search for the meaning of itemgetter, too) and reproducibility, since in python2.7 the order of the keys in a dictionary is not guaranteed and if the `locations` key would have come by accident before the `blockSize` key we would have had unexpected behavior.

Thanks again!

Reply:
Change applied!

Reply:
Haa, I haven't noticed that it could fail in python2.7 if keys were in a different order. Very good point!